avro/src/main/scala/magnolify/avro/AvroType.scala

```diff
@@ -64,14 +63,16 @@ sealed trait AvroField[T] extends Serializable { self =>
 
   @transient private lazy val schemaCache: concurrent.Map[ju.UUID, Schema] =
     concurrent.TrieMap.empty
-  protected def schemaString(cm: CaseMapper): String
+
+  protected def createSchema(cm: CaseMapper): Schema
   def schema(cm: CaseMapper): Schema =
-    schemaCache.getOrElseUpdate(cm.uuid, new Schema.Parser().parse(schemaString(cm)))
+    schemaCache.getOrElseUpdate(cm.uuid, createSchema(cm))
 
-  // Convert case class default `T` to Avro Java type
-  def makeDefault(p: Option[T])(cm: CaseMapper): Option[Any] = p.map(to(_)(cm))
-  // Fallback Avro Java type
-  val fallbackDefault: Any = null
+  // Convert default `T` to Avro schema default value
+  def makeDefault(d: T)(cm: CaseMapper): Any = to(d)(cm)
```

Comment:
This is a breaking change but should not be accessed by the users.
The signature wasn't nice, we should avoid passing the applicative  as parameter to map over it and return